tblgen/hs-generators.cc

```diff
@@ -536,7 +545,30 @@ pattern {0} loc {2:$[ ]} {3:$[ ]} {4:$[ ]} = {5}
                       make_range(type_binders),          // 2
                       make_range(operand_binders),       // 3
                       make_range(attr_pattern.binders),  // 4
-                      *operation);                       // 5
+                      *operation,                        // 5
+                      op.getOperationName());            // 6
+
+}
+
+std::string formatDescription(mlir::tblgen::Operator op) {
+  std::string description;
+  description = "\n" + op.getDescription().str();
+  size_t pos = 0;
+  while (description[pos] == '\n') ++pos;
+  size_t leading_spaces = 0;
```

Comment:
If you use StringRef, you can use it's utility methods to trim.

Reply:
Does it have a multiline trim? That's what I'm looking for here.

Reply:
Yes, StringRef trim (ltrim/rtrim) don't care about lines, it is just plain strings and they remove whitespace

Reply:
Yeah but I want to trim stuff that is embedded inside the string, not just leading or trailing whitespace. And also I want to only trim the prefix depth I infer form the first line, because sometimes indentation is meaningful and I want to preserve it.

Reply:
Gotcha, in the doc gen I added

raw_indented_ostream &mlir::raw_indented_ostream::reindent(StringRef str)

for some of these. But that follows more a streaming approach.

Reply:
I tried using the `raw_indented_ostream` here but I ended up with a bunch of missing symbol errors... I'll keep this for now, because it's not too long anyway.